src/api/core/ciphers.rs

```diff
@@ -238,6 +247,20 @@ pub fn update_cipher_from_data(
     nt: &Notify,
     ut: UpdateType,
 ) -> EmptyResult {
+    // Check that the client isn't updating an existing cipher with stale data.
+    if data.LastKnownRevisionDate.is_some() {
+        let dt = data.LastKnownRevisionDate.unwrap();
+        let res = NaiveDateTime::parse_from_str(&dt, "%+"); // ISO 8601 format
+        if res.is_ok() {
+            let dt = res.unwrap();
+            if cipher.updated_at.signed_duration_since(dt).num_seconds() > 1 {
+                err!("The client copy of this cipher is out of date. Resync the client and try again.")
+            }
+        } else {
+            warn!("error parsing LastKnownRevisionDate '{}': {}", dt, res.unwrap_err());
+        }
+    }
```

Comment:
This is more idiomatic rust code:

```suggestion
    if let Some(dt) = &data.LastKnownRevisionDate {
        match NaiveDateTime::parse_from_str(dt, "%+") {
            Ok(dt) if cipher.updated_at.signed_duration_since(dt).num_seconds() > 1 => {
                err!("The client copy of this cipher is out of date. Resync the client and try again.");
            }
            Ok(_) => (),
            Err(err) => warn!("error parsing LastKnownRevisionDate '{}': {}", dt, err),
        }
    }
```

Reply:
Thanks, I wasn't aware of conditionals on match arms. I amended to use a variant of your suggestion.